File: nats/node/nats_demo_publisher.js
const NATS = require('nats');
const nats = NATS.connect({ servers: ['nats://127.0.0.1:4222'] });

// EJEMPLO 1 Publicar un mensaje en JSON y después cerrar la conexión
const { random } = require('./random');
nats.publish('prices.daily', JSON.stringify({ MSFT: random(0, 100) }), () => {
  nats.close();
});

// EJEMPLO 2 Publicar dos mensajes en dos colas distintas y después cerrar la conexión
// nats.publish('foo.bar', `FIRST ${new Date().toISOString()}`, () => {
//   nats.publish('foo.yeah.thing', `SECOND ${new Date().toISOString()}`, () => {
//     nats.close();
//   });
// });

File: nats/python/nats_demo_publisher.py
import asyncio
import signal

from datetime import datetime

from nats.aio.client import Client as NATS
from nats.aio.errors import ErrConnectionClosed, ErrTimeout, ErrNoServers

async def main():
  nc = NATS()
  await nc.connect(servers=["nats://127.0.0.1:4222"])

  await nc.publish("foo.thing", b'Hello!')
  message = 'Current date: at {now}'.format(now=datetime.now().isoformat())
  await nc.publish("foo.bar", message.encode())

if __name__ == '__main__':
  asyncio.run(main())
